### PR TITLE
fix(send_kcidb): Do not crash if optional field is missing

### PR DESCRIPTION
**Pull request by nuclearcat**
Some fields might be not present, as they are optional in model. Also add this field in custom checkout, even it is not mandatory, it is better to have it for consistency.

### CODE REVIEW

File: src/send_kcidb.py [hunk 1]
```diff
@@ -141,11 +141,11 @@ def _parse_checkout_node(self, origin, checkout_node):
...
                 checkout_node['data']['kernel_revision']['branch'],
             'git_commit_tags':
-                checkout_node['data']['kernel_revision']['commit_tags'],
+                checkout_node['data']['kernel_revision'].get('commit_tags'),
             'git_commit_message':
-                checkout_node['data']['kernel_revision']['commit_message'],
+                checkout_node['data']['kernel_revision'].get('commit_message'),
             'git_repository_branch_tip':
-                checkout_node['data']['kernel_revision']['tip_of_branch'],
+                checkout_node['data']['kernel_revision'].get('tip_of_branch', False),
```
Comment (JenySadadia) [inline src/send_kcidb.py:148]: We can keep it `None` instead of `False` if the flag is not set.

File: src/lava_callback.py [hunk 2]
```diff
@@ -600,7 +600,8 @@ async def checkout(data: ManualCheckout, request: Request,
                 "tree": treename,
                 "branch": branch,
                 "commit": commit,
-                "url": treeurl
+                "url": treeurl,
+                "tip_of_branch": False
```
Comment (JenySadadia) [inline src/lava_callback.py:604]: Custom checkout can also have a commit from the tip of the branch. Why is it set `False`?

Comment (JenySadadia) [inline src/lava_callback.py:604]: Custom checkout can also have a commit from the tip of the branch. Why is it set `False`?

Comment (nuclearcat) [reply src/lava_callback.py:604]: We cannot trust user to indicate if it is TOT, but to test it separately on our side will need additional logic to implement (or fix existing one). 
We might check it later, as it is not critical right now, custom checkout events should not be sent to kcidb anyway (yet).

File: src/send_kcidb.py (hunk 1 above)
Comment (nuclearcat) [reply src/send_kcidb.py:148]: Ok

File: src/lava_callback.py (hunk 2 above)
Comment (JenySadadia) [reply src/lava_callback.py:604]: I don't think we rely much on the fact if the commit is at the tip of branch or not. We are just using that data to send to KCIDB.
I think we can set its value from user's input and set it to `None` if not provided.

Comment (nuclearcat) [reply src/lava_callback.py:604]: It worked fine, your code updated it to true, if it is tipoftree, so we can leave as is

**Comment by nuclearcat** [comment]
Tested on staging

**Pull request merged**